### ISSUE TITLE
VariableDeclarationAfterUsage gives false negative if var declared on same line after var reference

### ISSUE THREAD

**Issue by EliZukowski16** [SonarSource/SonarJS#961]
**I want to report a bug.**

For the `VariableDeclarationAfterUsage` rule, if a variable declaration occurs on the same line, but **after** a variable is referenced, SonarJS will fail to flag the initial variable reference as a violation of this rule.

Looking at the [implementation of the check](https://github.com/SonarSource/SonarJS/blob/master/javascript-checks/src/main/java/org/sonar/javascript/checks/VariableDeclarationAfterUsageCheck.java) a possible solution that I can see is adding some additional logic to the `LineComparator`  that checks the column position of the two usages if the line numbers are the same.

**SonarJS version:** 3.2.0.5506<br>
**SonarQube version:** 5.6.6<br>

**Rule key:** VariableDeclarationAfterUsage

**Reproducer** 
```javascript
if(foo === "undefined") { var foo = {}) //False Negative

if(foo === "undefined") { //correctly flags as violating rule
    var foo = {} 
}
```

**Comment by ilia-kebets-sonarsource** [comment]
Both errors get flagged by the rule. I believe we can close this one.
